great_expectations/datasource/base_data_asset.py

```diff
@@ -0,0 +1,80 @@
+import logging
+from abc import ABC
+from dataclasses import dataclass
+from typing import Dict, List, Optional
+from typing_extensions import Self
+
+from great_expectations.core.batch import Batch, BatchRequest
+from great_expectations.datasource.misc_types import NewBatchRequestBase
+# from great_expectations.datasource.new_new_new_datasource import NewNewNewDatasource
+from great_expectations.validator.validator import Validator
+
+logger = logging.getLogger(__name__)
+
+class BaseDataAsset:
+    def __init__(
+        self,
+        datasource,  # Should be of type: V15Datasource
+        name: str,
+        batch_identifiers: List[str],
+    ) -> None:
+        self._datasource = datasource
+        self._name = name
+        self._batch_identifiers = batch_identifiers
+
+    @property
+    def datasource(self): # Should be: -> NewNewNewDatasource:
+        return self._datasource
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @property
+    def batch_identifiers(self) -> List[str]:
+        return self._batch_identifiers
+
+    def set_name(self, name: str):
+        """Changes the DataAsset's name.
+
+        Note: This method is intended to be called only from Dataasource's rename_asset method.
+        This will keep the name of the asset in sync with the key in the Datasource's _asset registry.
+        """
+        self._name = name
+
+    def update_configuration(self, **kwargs) -> Self:
+        raise NotImplementedError
+        
+    def list_batches(self) -> List[NewBatchRequestBase]:
+        raise NotImplementedError
+
+    def get_batch(self, *args, **kwargs) -> Batch:
```

Comment:
These method signatures are not helpful ... how do I get a batch?

Reply:
I know you probably felt hurt and betrayed to see args and kwargs in method signatures, but hear me out.

Assuming we want to go down the road of Datasource proliferation---which I think is the right decision---then we're going to have:
1. many Datasources
2. each with a `get_batch` method,
3. each taking a different set of arguments.

In that world, it makes sense for the base class `get_batch` method to have an extremely general method declaration, so that other classes can have more specific signatures.

Note: The base class method docstring should explain all this. Then the child class methods can have actual arguments and informative docstrings. IMO, that's the best division of labor here.

Note: I also considered the alternative of having each child class declare its own `get_batch_from_*something*` method(s). This would avoid the args/kwargs issues, but it has other problems.